pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/security/xml/ApexCRUDViolation.xml

```diff
@@ -376,6 +376,45 @@ public class Foo {
         ]]></code>
     </test-code>
 
+    <test-code>
+        <description>Flagged as Critical With No Mode</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class User {
+    public void coverAllCasesWithTest() {
+        Contact c;
+		c = [SELECT Name FROM Contact];
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>User Mode</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class User {
+    public void coverAllCasesWithTest() {
+        Contact c;
+		c = [SELECT Name FROM Contact WITH USER_MODE];
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>System Mode, gives warning because it ignores CRUD but explicitly</description>
+        <expected-problems>0</expected-problems>
```

Comment:
```suggestion
        <expected-problems>1</expected-problems>
        <expected-messages>
            <message>This CRUD statement uses explicit system mode</message>
        </expected-messages>
```

PMD can't create violations with different priorities (e.g. error / warning) for the same rule. So, we can't create a warning, but we can create a violation as well for this case. We can e.g. override the violation message to make it clear, that it is a slightly different problem that PMD identified.

We could also introduce a rule property which controls, whether usages of system modes should be flagged or not. One can argue that using system mode, the developer explicitly decided to use elevated permissions and it is not default anymore. So, not everybody might want to see violations when they explicitly use system mode. Hence, a property might be useful (e.g. "reportExplicitSystemMode").